scipy/stats/qmc.py

```diff
@@ -0,0 +1,229 @@
+# -*- coding: utf-8 -*-
+r"""
+====================================================
+Quasi-Monte Carlo submodule (:mod:`scipy.stats.qmc`)
+====================================================
+
+.. currentmodule:: scipy.stats.qmc
+
+This module provides Quasi-Monte Carlo generators and associated helper
+functions.
+
+
+Quasi-Monte Carlo
+=================
+
+Engines
+-------
+
+.. autosummary::
+   :toctree: generated/
+
+   QMCEngine
+   Sobol
+   Halton
+   OrthogonalLatinHypercube
+   LatinHypercube
+   OptimalDesign
+   NormalQMC
+   MultivariateNormalQMC
+
+Helpers
+-------
+
+.. autosummary::
+   :toctree: generated/
+
+   discrepancy
+   scale
+   multinomial_qmc
+
+
+Introduction to Quasi-Monte Carlo
+=================================
+
+Quasi-Monte Carlo (QMC) methods [1]_, [2]_, [3]_ provide an
+:math:`n \times dim` matrix of numbers in :math:`[0,1]`. They can be used in
+place of n points from the :math:`U[0,1]^{dim}` distribution. Compared to
+random points, QMC points are designed to have fewer gaps and clumps. This is
+quantified by discrepancy measures [4]_. From the Koksma-Hlawka
+inequality [5]_ we know that low discrepancy reduces a bound on
+integration error. Averaging a function :math:`f` over :math:`n` QMC points
+can achieve an integration error close to :math:`O(n^{-1})` for well
+behaved functions [2]_.
+
+Most QMC constructions are designed for special values of :math:`n`
+such as powers of 2 or large primes. Changing the sample
+size by even one can degrade their performance, even their
+rate of convergence [6]_. For instance :math:`n=100` points may give less
+accuracy than :math:`n=64` if the method was designed for :math:`n=2^m`.
+
+Some QMC constructions are extensible in n: we can find
+another special sample size :math:`n' > n` and often an infinite
+sequence of increasing special sample sizes. Some QMC
+constructions are extensible in :math:`dim`: we can increase the dimension,
+possibly to some upper bound, and typically without requiring
+special values of d. Some QMC methods are extensible in
+both :math:`n` and :math:`dim`.
+
+QMC points are deterministic. That makes it hard to estimate
+the accuracy of averages. Randomized QMC (RQMC) [7]_
+points are constructed so that each point is individually :math:`U[0,1]^{dim}`
+while collectively the n points retain their low discrepancy.
+One can make :math:`R` independent replications of RQMC points to
+see how stable a computation is. From :math:`R` independent values
+a t test (or bootstrap t test [8]_) then gives approximate confidence
```

Comment:
What do you mean?

Reply:
gh-4824 implements a permutation t-test using `np.random.permutation`. It sounded like in the future it might be possible to add the option to use features of this submodule instead of `np.random`?